contracts/pol/CarbonPOL.sol

```diff
@@ -164,17 +210,19 @@ contract CarbonPOL is ICarbonPOL, Upgradeable, ReentrancyGuardUpgradeable, Utils
     /**
      * @inheritdoc ICarbonPOL
      */
-    function expectedTradeReturn(Token token, uint128 ethAmount) external view validToken(token) returns (uint128) {
+    function expectedTradeReturn(Token token, uint128 tradeInput) external view validToken(token) returns (uint128) {
         Price memory currentPrice = tokenPrice(token);
         // revert if price is not valid
         _validPrice(currentPrice);
-        // multiply the token amount by the eth amount / total eth amount ratio to get the actual tokens received
-        uint128 tokenAmount = MathEx.mulDivF(currentPrice.tokenAmount, ethAmount, currentPrice.ethAmount).toUint128();
+        // calculate the trade return based on the current price and token
+        uint128 tradeReturn = MathEx
+            .mulDivF(currentPrice.targetAmount, tradeInput, currentPrice.sourceAmount)
+            .toUint128();
         // revert if not enough token balance
-        if (tokenAmount > token.balanceOf(address(this))) {
+        if (tradeReturn > token.balanceOf(address(this))) {
```

Comment:
balanceOf isn't correct for ETH as it's limited by the amount available for trading.
I suggest a utility function (`amountAvailableForTrading`) that calls balanceOf for tokens, or returns the current ETH sale amount or ETH.

Reply:
Fixed by adding the `amountAvailableForTrading` utility function